pkg/controller/installation/core_controller.go

```diff
@@ -1214,9 +1214,16 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	// If we're on OpenShift on AWS render a Job (and needed resources) to
 	// setup the security groups we need for IPIP, BGP, and Typha communication.
 	if openShiftOnAws {
+		// Detect if this cluster is an OpenShift HPC hosted cluster, as AWS
+		// security group setup is different in this case.
+		isHostedOpenShift, err := utils.IsHostedOpenShift(ctx, r.config)
```

Comment:
This is a little bit of a bad pattern - this is going to cause an API call on every reconcile, since it's instantiating a direct non-cached client.

We should instead register a watch for this object, and use the runtime client to read it from the cache. This will avoid the extra API calls.